enum_methods.rb

```diff
@@ -0,0 +1,146 @@
+module Enumerable
+  def my_each
+    if block_given?
+      for i in self
+        yield(i)
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_each_with_index
+    count = 0
+    if block_given?
+      for i in self
+        yield(i, count)
+        count += 1
+      end
+    else
+      puts 'No Block Given'
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      result = {}
+      my_each { |key, value| result[key] = value if yield(key, value) == true }
+    else
+      result = []
+      my_each { |value| result.push(value) if yield(value) == true }
+    end
+    result
+  end
+
+  def my_all?
+    count = 0
+    my_each { |value| count += 1 if yield(value) == true }
+    count == length
+  end
```

Comment:
- [ ]  my_all? returns false if the block returns false or nil::range

- [ ]  my_all? when no block or argument is given returns true when none of the collection members are false or nil

- [ ] my_all? when no block or argument is given returns false when one of the collection members are false or nil

- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Integer

- [ ] my_all? when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric